src/context.rs

```diff
@@ -37,12 +37,14 @@ impl<N: Name + fmt::Debug> error::Error for UnificationError<N> {
 #[derive(Debug, Clone, PartialEq, Eq)]
 pub struct Context<N: Name = &'static str> {
     pub(crate) substitution: IndexMap<Variable, Type<N>>,
+    pub(crate) cache: HashMap<Variable, Type<N>>,
```

Comment:
Especially given the similarity to `substitution` (they’re both maps from variable to type), this should at least get a better name, e.g. `path_compression_cache` and maybe even a comment explaining it (if you take my suggestion from the other comment about always using path compression, maybe you can move your detailed explanation into here)

Reply:
Done!